src/document.h

```diff
@@ -686,8 +686,8 @@ struct Document {
             int maxdepth = 0, leaves = 0;
             root->MaxDepthLeaves(0, maxdepth, leaves);
             if (maxdepth > 1)
-                return _(L"Cannot export grid that is not flat (zoom the view to the desired grid, "
-                         L"and/or use Flatten).");
+                return _(L"Cannot export grid that is not flat (zoom the view to the desired grid, \

```

Comment:
Thanks you for your comprehensive investigation to fix this. I am not sure I like this fix however, it makes source code hard to read and maintain, and no-one writes strings like this in source code. There has to be a better way. If there is no way for the tools to recognize multi-line strings of this kind correctly (why? surely recent versions of these tools must recognize modern C++?) then simply making them into single lines is a better solution for the moment, as it doesn't hurt code readability quite as much.

Reply:
> Thanks you for your comprehensive investigation to fix this.

You're welcome.

I don't know whether there's a way for `xgettext` to detect multi-line strings of this kind correctly. The [documentation](https://www.gnu.org/software/gettext/manual/html_node/xgettext-Invocation.html#Output-details) states that

> This implementation of `xgettext` is able to process a few awkward cases,
> like strings in preprocessor macros, ANSI concatenation of adjacent strings,
> and escaped end of lines for continued strings.

but that did not work with the binaries for version 0.21 (the latest version is 0.21.1) I downloaded from the [site](https://mlocati.github.io/articles/gettext-iconv-windows.html) mentioned in `TS/translations/readme_translations.txt`.

I too think that there should be a better way, but couldn't find any, and while I read on the subject I found the [Audacity wiki page](https://wiki.audacityteam.org/wiki/Translating_Audacity/Writing_Translatable_Code#Dealing_with_Long_Strings) I mentioned in my [first comment](https://github.com/aardappel/treesheets/pull/281#issue-1441202645). It states that

> When strings get longer and span several lines we place a '\' at the end
> of each line, and continue immediately at the start of the next line.
> We have tried various other ways of dealing with long strings, but the
> other ways we've tried either don't work when compiling with Unicode,
> or defeat detection by xgettext. 

So I chose to do the modification that didn't use long lines.

I'll push another commit that makes those ugly multiline strings into single lines.

Reply:
That is unfortunate.. not sure how other projects deal with this..